stores/governance/src/main/java/com/bloxbean/cardano/yaci/store/governance/service/LocalGovStateService.java

```diff
@@ -0,0 +1,345 @@
+        });
+
+        if (!localCommitteeMemberEntities.isEmpty()) {
+            localCommitteeMemberStorage.saveAll(localCommitteeMemberEntities);
+        }
+    }
+
+    private void handleTreasuryWithdrawals(List<GovActionId> enactedGovActionIds, Integer currentEpoch, Long slot) {
+        List<GovActionProposal> enactedTreasuryWithdrawalProposals = govActionProposalStorage.findByGovActionIds(enactedGovActionIds)
+                .stream()
+                .filter(govActionProposal -> govActionProposal.getType() == GovActionType.TREASURY_WITHDRAWALS_ACTION)
+                .toList();
+
+        localTreasuryWithdrawalStorage.saveAll(
+                enactedTreasuryWithdrawalProposals.stream()
+                        .flatMap(govActionProposalEntity -> {
+                            JsonNode actionDetail = govActionProposalEntity.getDetails();
+                            JsonNode withdrawalsNode = actionDetail.get("withdrawals");
+                            List<LocalTreasuryWithdrawal> treasuryWithdrawalEntities = new ArrayList<>();
+                            withdrawalsNode.fields().forEachRemaining(entry -> {
+                                String addressHash = entry.getKey();
+                                BigInteger amount = entry.getValue().bigIntegerValue();
+                                Address address = new Address(HexUtil.decodeHexString(addressHash));
+                                var entity = LocalTreasuryWithdrawal.builder()
+                                        .govActionTxHash(govActionProposalEntity.getTxHash())
+                                        .govActionIndex((int) govActionProposalEntity.getIndex())
+                                        .epoch(currentEpoch)
+                                        .amount(amount.longValue())
+                                        .address(address.getAddress())
+                                        .slot(slot)
+                                        .build();
+                                treasuryWithdrawalEntities.add(entity);
+                            });
+
+                            return treasuryWithdrawalEntities.stream();
+                        }).toList());
+    }
+
+    private void handleHardForkInitiation(List<GovActionId> enactedGovActionIds, Integer currentEpoch, Long slot) {
+        List<GovActionProposal> enactedHardForkInitProposals = govActionProposalStorage.findByGovActionIds(enactedGovActionIds)
+                .stream()
+                .filter(govActionProposalEntity -> govActionProposalEntity.getType() == GovActionType.HARD_FORK_INITIATION_ACTION)
+                .toList();
+
+        localHardForkInitiationStorage.saveAll(
+                enactedHardForkInitProposals.stream()
+                        .map(govActionProposal -> {
+                            JsonNode actionDetail = govActionProposal.getDetails();
+                            JsonNode protocolVersionNode = actionDetail.get("protocolVersion");
```

Comment:
I think in this case we don't need check null.
maybe we can't submit proposal with govaction detail is null, and it can't be ratified if its detail is null